ui/components/src/model/units/bytes.ts

```diff
@@ -60,17 +61,16 @@ export function formatBytes(bytes: number, options: BytesUnitOptions) {
       }
     }
 
-    const formatter = Intl.NumberFormat('en-US', formatterOptions);
-    return formatter.format(bytes);
+    return Intl.NumberFormat('en-US', formatterOptions).format(bytes);
   }
 
   // numbro is able to add units like KB, MB, GB, etc. correctly
   return numbro(bytes).format({
     output: 'byte',
     base: 'decimal',
     spaceSeparated: true,
-    mantissa: decimal_places ?? DEFAULT_NUMBRO_MANTISSA,
-    trimMantissa: true,
-    optionalMantissa: true,
+    mantissa: hasDecimalPlaces(decimal_places) ? decimal_places : DEFAULT_NUMBRO_MANTISSA,
+    trimMantissa: !hasDecimalPlaces(decimal_places),
+    optionalMantissa: !hasDecimalPlaces(decimal_places),
```

Comment:
`trimMantissa` trims trailing 0s
`optionalMantissa` excludes the decimals places if they're all 0s

We don't want this behavior if `decimal_places` is set

Reply:
Might be worth putting this in a comment above these for future us.

Reply:
Yess